packages/app/tasks/set-programs.js

```diff
@@ -0,0 +1,167 @@
+
+            for (const program of programsBatch) {
+                const {
+                    poolToken,
+                    networkToken,
+                    reserveToken,
+                    startTime,
+                    endTime,
+                    networkTokenRate,
+                    reserveTokenRate,
+                    rewardRate
+                } = program;
+
+                const data = await web3Provider.call(contracts.StakingRewardsStore.methods.poolProgram(poolToken));
+                const actualStartTime = data[0];
+                if (actualStartTime > 0) {
+                    trace('Scheduling existing program for removal', arg('poolToken', poolToken));
+
+                    poolTokensToRemove.push(poolToken);
+                }
+
+                poolTokens.push(poolToken);
+                reserveTokens.push([networkToken, reserveToken]);
+                rewardShares.push([networkTokenRate, reserveTokenRate]);
+                startTimes.push(startTime);
+                endTimes.push(endTime);
+                rewardRates.push(rewardRate);
+            }
+
+            if (poolTokens.length === 0) {
+                continue;
+            }
+
+            for (const poolToken of poolTokensToRemove) {
+                info('Removing already participating program', arg('poolToken', poolToken));
+
+                const tx = await web3Provider.send(contracts.StakingRewardsStore.methods.removePoolProgram(poolToken));
+                totalGas += tx.gasUsed;
+            }
+
+            const tx = await web3Provider.send(
+                contracts.StakingRewardsStore.methods.addPastPoolPrograms(
+                    poolTokens,
+                    reserveTokens,
+                    rewardShares,
+                    startTimes,
+                    endTimes,
+                    rewardRates
+                )
+            );
+            totalGas += tx.gasUsed;
+        }
+
+        info('Finished adding new pools times', arg('totalGas', totalGas));
+    };
+
+    const verifyPrograms = async (programs) => {
+        info('Verifying programs');
+
+        for (const program of programs) {
+            const {
+                poolToken,
+                networkToken,
+                reserveToken,
+                startTime,
+                endTime,
+                networkTokenRate,
+                reserveTokenRate,
+                rewardRate
+            } = program;
+
+            trace('Verifying program', arg('poolToken', poolToken));
+
+            const data = await web3Provider.call(contracts.StakingRewardsStore.methods.poolProgram(poolToken));
+
+            const actualStartTime = data[0];
+            const actualEndTime = data[1];
+            const actualRewardRate = data[2];
+            const [actualNetworkToken, actualReserveToken] = data[3];
+            const [actualNetworkTokenRate, actualReserveTokenRate] = data[4];
+
+            if (actualStartTime != startTime) {
```

Comment:
don't we need `new BN()` here as well?

Reply:
Ah, it should work without it, but better safe than sorry